library/block/block_content_registry/src/main/java/org/quiltmc/qsl/block/content/registry/mixin/EnchantmentScreenHandlerMixin.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.block.content.registry.mixin;
+
+import it.unimi.dsi.fastutil.objects.ObjectIterators;
+import net.minecraft.block.Block;
+import net.minecraft.block.EnchantingTableBlock;
+import net.minecraft.item.ItemStack;
+import net.minecraft.screen.EnchantmentScreenHandler;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.quiltmc.qsl.block.content.registry.api.BlockContentRegistries;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import java.util.Iterator;
+import java.util.List;
+
+@Mixin(EnchantmentScreenHandler.class)
+public class EnchantmentScreenHandlerMixin {
+	@Redirect(method = "m_mpsetdhw", at = @At(value = "INVOKE", target = "Ljava/util/List;iterator()Ljava/util/Iterator;"))
```

Comment:
Due to the migration to Intermediary QM, you'll want to change these two in order to not crash in-prod:
```suggestion
	@Redirect(method = "m_mpsetdhw(Lnet/minecraft/item/ItemStack;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V", at = @At(value = "INVOKE", target = "Ljava/util/List;iterator()Ljava/util/Iterator;"))
```